cfg/validate.go

```diff
@@ -28,10 +29,25 @@ func isValidLogRotateConfig(config *LogRotateLoggingConfig) error {
 	return nil
 }
 
+func isValidURL(u string) (string, error) {
```

Comment:
The function name is misleading. It sounds like it is only validating the given URI, but it's also trying to parse it and return the successfully parsed url or error. How about 

```suggestion
func decodedURL(u string) (string, error) {
```

or 

```suggestion
func parsedURL(u string) (string, error) {
```

Reply:
@kislaykishore and I had a discussion and we decided to duplicate this in the Rationalise() function keeping the responsibility of validate and rationalise separate. I've taken these suggestions in the rationalise part.